### ISSUE TITLE
[Maestro version 1.36.0] Failed to parse file when using subflows with maestro cloud CLI

### ISSUE THREAD

**Issue by paulsweeting** [mobile-dev-inc/maestro#1791]
**Describe the bug**
Upload requests fail to maestro cloud when using the CLI if the test contains sub-flows:

`- runFlow: subflows/onboarding-android.yaml`

**To Reproduce**
Follow https://cloud.mobile.dev/getting-started/quickstart in login into CLI and download samples

1. Start the simulator/emulator
2. Run `maestro cloud sample.zip ios-advanced-flow.yaml OR maestro cloud sample.apk android-advanced-flow.yaml`
3. The Flow fails with error:
```
Upload request failed (400): Syntax error: Failed to parse file: /./ios-advanced-flow.yaml
Flow file does not exist: jar:file:///tmpFiles/workspaces/workflow197377531878186231.zip!/./subflows/onboarding-ios.yaml
```

**Expected behavior**
It works as expected with simple flows that do NOT contain subflow e.g. maestro cloud sample.zip ios-flow.yaml

**Screenshots**

**Environment information (please complete the following information):**

- Maestro version [1.36.0]
- Platform: [Android && iOS]
- Framework: [Native iOS/Android]
- Device model and OS version: [N/A]
- Cloud 
- Host [Mac M3]

**Comment by bartekpacia** [comment]
Hi @paulsweeting, thanks for creating the issue and sorry for the problems.

Could you share a simple, example flows just so that we can reproduce this easily?

**Comment by paulsweeting** [comment]
@bartekpacia - Yeap, I'm just using the out of the box example flows when you run `maestro download-samples`

`appId: org.wikipedia
tags:
  - advanced
---
- runFlow: subflows/onboarding-android.yaml
- tapOn:
    id: "org.wikipedia:id/search_container"
- runScript: scripts/getSearchQuery.js
- inputText: ${output.result}
- assertVisible: ${output.result}`

**Comment by bartekpacia** [comment]
Thanks for the report, I confirm it occurs.

**Comment by bartekpacia** [comment]
Looks like this is expected behavior. When using subflows, you have to upload the whole workspace. This is explained in the docs [here](https://cloud.mobile.dev/getting-started/quickstart):

<img width="817" alt="Screenshot 2024-07-11 at 19 22 35" src="https://github.com/mobile-dev-inc/maestro/assets/40357511/0abb86aa-6799-4886-a7af-ff534c9cf923">

I agree a better job should be done here, to inform users about this. Maestro CLI should maybe say:

```
Your flow includes a subflow. To run on Maestro Cloud, you need to upload the whole workspace instead:

<example invocation>
```

See also:
- https://github.com/mobile-dev-inc/maestro/issues/1696

**Comment by paulsweeting** [comment]
Thank you @bartekpacia I can confirm that following works and runs all flows in the samples folder

`maestro cloud samples/sample.zip samples`

However, we want to be able to single advanced flows. All of our tests use subflows. When writing a new test, we'd like the ability to run the test via the `maestro cloud` CLI to confirm it runs and passes before raising the PR, where all tests run as part of GitHub actions. 

For example, using the samples, we want to be able to run:

`maestro cloud samples/sample.zip samples/ios-advanced-flow.yaml`

Which currently gives the same error:

```
Upload request failed (400): Syntax error: Failed to parse file: /./ios-advanced-flow.yaml
Flow file does not exist: jar:file:///tmpFiles/workspaces/workflow17464117397033909285.zip!/./subflows/onboarding-ios.yaml
```

From what you've said this isn't possible at the moment. Is there a feature request open for this already? Or any suggested workarounds?

**Comment by bartekpacia** [comment]
No, afaik, it's not possible as well right now. I see why you need this. Sorry for the problems.

> Is there a feature request open for this already?

Let's consider this issue to be a feature request.

**Comment by paulsweeting** [comment]
Just an update for anyone looking for a workaround. To run single advanced tests you can use tags. For example, add the tag '`single-test-run`' to the test:

```
appId: org.wikimedia.wikipedia
tags:
  - advanced
  - single-test-run
```

Then pass the tag through using `--include-tags`

`maestro cloud samples/sample.zip samples --include-tags=single-test-run`

Which runs the single test:

```
[Passed] ios-advanced-flow (1m 21s)
1/1 Flow Passed
```

**Comment by YakirLavi1** [comment]
@paulsweeting thanks for the workaround, much appriciated.

Any update about this issue?
